ambry-clustermap/src/test/java/com.github.ambry.clustermap/MockPartitionId.java

```diff
@@ -66,7 +66,7 @@ public MockPartitionId(long partition, String partitionClass, List<MockDataNodeI
 
   @Override
   public List<ReplicaId> getReplicaIds() {
-    return replicaIds;
+    return new ArrayList<>(replicaIds);
```

Comment:
If your intention was just to disallow modifications by the caller, you could instead wrap this using `Collections.unmodifiableList()`.

Reply:
We just want to have a copy of this data. Otherwise, different callers may shuffle the same reference in `SimpleOperationTracker`.